pkg/aggregator/time_sampler_test.go

```diff
@@ -532,6 +534,39 @@ func BenchmarkTimeSampler(b *testing.B) {
 	benchWithTagsStore(b, benchmarkTimeSampler)
 }
 
+func BenchmarkTimeSamplerWithLimiter(b *testing.B) {
+	sample1 := metrics.MetricSample{
+		Name:       "foo",
+		Value:      1,
+		Mtype:      metrics.GaugeType,
+		Tags:       []string{"foo", "bar"},
+		SampleRate: 1,
+		Timestamp:  12345.0,
+	}
+	sample2 := metrics.MetricSample{
+		Name:       "bar",
+		Value:      1,
+		Mtype:      metrics.GaugeType,
+		Tags:       []string{"foo", "bar"},
+		SampleRate: 1,
+		Timestamp:  12345.0,
+	}
+
+	for limit := range []int{0, 1, 2, 3} {
+		store := tags.NewStore(false, "test")
+		limiter := limiter.New(limit, "pod", []string{"pod"})
+		fmt.Println(limiter)
+		sampler := NewTimeSampler(TimeSamplerID(0), 10, store, limiter, "host")
+
+		b.Run(fmt.Sprintf("limit=%d", limit), func(b *testing.B) {
```

Comment:
Would it be interesting to see benchmarks with and without a limiter?

Reply:
Yes, and if I understand you correctly, that's already covered by the benchmark: the zero limit corresponds to the limiter being as disabled as the code can support (limiter.New returns nil in that case already).